homeassistant/components/xiaomi_miio/gateway.py

```diff
@@ -56,12 +58,22 @@ async def async_connect_gateway(self, host, token):
             )
 
             # get the connected sub devices
-            if (
-                use_cloud
-                and cloud_username is not None
-                and cloud_password is not None
-                and cloud_country is not None
-            ):
+            if use_cloud or self._gateway_info.model == GATEWAY_MODEL_EU:
+                if (
+                    cloud_username is None
+                    or cloud_password is None
+                    or cloud_country is None
+                ):
+                    # trigger re-auth flow
+                    self._hass.async_create_task(
```

Comment:
We can raise `ConfigEntryAuthFailed` to have the config entries manager start the re-auth flow for us automatically.

Reply:
@MartinHjelmare will that also include the config_entry data?

Reply:
Already tested it, it works.
Thanks for the hint!